lms/djangoapps/discussion_api/tests/test_views.py

```diff
@@ -1566,3 +1759,28 @@ def test_pagination(self):
             404,
             {"developer_message": "Page not found (No results on this page)."}
         )
+
+    def test_profile_image_requested_field(self):
+        """
+        Tests all comments retrieved have user profile image details if called in requested_fields
+        """
+        self.register_get_user_response(self.user)
+        cs_comment_child = self.make_comment_data('test_child_comment', self.comment_id, children=[])
+        cs_comment = self.make_comment_data(self.comment_id, None, [cs_comment_child])
+        cs_thread = make_minimal_cs_thread({
+            'id': self.thread_id,
+            'course_id': unicode(self.course.id),
+            'children': [cs_comment],
+        })
+        self.register_get_thread_response(cs_thread)
+        self.register_get_comment_response(cs_comment)
+        self.create_profile_image(self.user, get_profile_image_storage())
+
+        response = self.client.get(self.url, {'requested_fields': 'profile_image'})
+        self.assertEqual(response.status_code, 200)
+        response_comments = json.loads(response.content)['results']
+
+        for response_comment in response_comments:
+            expected_profile_data = self.get_expected_user_profile(response_comment['author'])
+            response_users = response_comment['users']
+            self.assertEqual(expected_profile_data, response_users[response_comment['author']])
```

Comment:
@jcdyer How about we create a new file to handle all additional optional fields related functionality.
This would move 
`_get_user_profile_dict(), _user_profile(), _get_users(), _include_profile_image(), _add_additional_response_fields()` 
and any other future fields related functionality to a new file; may be named as `requested_fields.py`.